src/HydraAuction/Tx/FeeEscrow.hs

```diff
@@ -0,0 +1,118 @@
+module HydraAuction.Tx.FeeEscrow (
+  distributeFee,
+) where
+
+-- Prelude imports
+import Prelude
+
+-- Haskell imports
+import Control.Monad (void)
+
+-- Plutus imports
+import Plutus.V1.Ledger.Address (pubKeyHashAddress)
+import Plutus.V2.Ledger.Api (PubKeyHash)
+
+-- Cardano node imports
+import Cardano.Api.UTxO qualified as UTxO
+
+-- Hydra imports
+import Hydra.Cardano.Api (
+  Lovelace (..),
+  Value,
+  lovelaceToValue,
+  txOutValue,
+  valueToLovelace,
+  pattern ReferenceScriptNone,
+  pattern TxMintValueNone,
+  pattern TxOut,
+  pattern TxOutDatumNone,
+ )
+
+-- Hydra auction imports
+import HydraAuction.OnChain (AuctionScript (..))
+import HydraAuction.Tx.Common (
+  scriptPlutusScript,
+  scriptUtxos,
+ )
+import HydraAuction.Types (
+  AuctionTerms (..),
+  FeeEscrowRedeemer (..),
+ )
+import HydraAuctionUtils.L1.Runner (L1Runner)
+import HydraAuctionUtils.Monads (
+  fromPlutusAddressInMonad,
+  logMsg,
+ )
+import HydraAuctionUtils.Monads.Actors (
+  actorTipUtxo,
+  addressAndKeys,
+ )
+import HydraAuctionUtils.Tx.AutoCreateTx (
+  AutoCreateParams (..),
+  autoSubmitAndAwaitTx,
+ )
+import HydraAuctionUtils.Tx.Build (
+  mkInlinedDatumScriptWitness,
+ )
+import HydraAuctionUtils.Tx.Utxo (
+  filterAdaOnlyUtxo,
+ )
+
+distributeFee :: AuctionTerms -> L1Runner ()
+distributeFee terms = do
+  logMsg "Distributing fees to delegates"
+
+  (actorAddress, _, actorSk) <- addressAndKeys
+
+  actorMoneyUtxo <- filterAdaOnlyUtxo <$> actorTipUtxo
+
+  feeEscrowUtxo <- scriptUtxos FeeEscrow terms
+
+  feeEscrowValue <- case UTxO.pairs feeEscrowUtxo of
+    [(_, txOut)] -> pure $ txOutValue txOut
```

Comment:
There is a `scriptSingleUtxo` function for that.

Reply:
I don't think it work as nicely in this case, as I want both the reference to `feeEscrowUtxo` and to look at its value

Reply:
It returns `Maybe (TxIn, TxOut CtxUTxO)` which seems the same that you use.

Probably the name `..Utxo` is confusing, but I still do not found how to name `(TxIn, TxOut)` pair better. `Utxo` term being overloaded is awful, and fixing better vocabluary would be cool, but I do not know how to name it better.

cc @GeorgeFlerovsky